sys/net/gnrc/network_layer/ipv6/netif/gnrc_ipv6_netif.c

```diff
@@ -105,15 +105,23 @@ static ipv6_addr_t *_add_addr_to_entry(gnrc_ipv6_netif_t *entry, const ipv6_addr
     }
     else {
         if (!ipv6_addr_is_link_local(addr)) {
-#ifdef MODULE_GNRC_NDP_ROUTER
+#if defined(MODULE_GNRC_NDP_ROUTER) || defined(MODULE_GNRC_SIXLOWPAN_ND_ROUTER)
             /* New prefixes MAY allow the router to retransmit up to
              * GNRC_NDP_MAX_INIT_RTR_ADV_NUMOF unsolicited RA
              * (see https://tools.ietf.org/html/rfc4861#section-6.2.4) */
             if ((entry->flags & GNRC_IPV6_NETIF_FLAGS_ROUTER) &&
                 (entry->flags & GNRC_IPV6_NETIF_FLAGS_RTR_ADV)) {
-                entry->rtr_adv_count = GNRC_NDP_MAX_INIT_RTR_ADV_NUMOF;
                 mutex_unlock(&entry->mutex);    /* function below relocks mutex */
+#ifdef MODULE_GNRC_NDP_ROUTER
+                entry->rtr_adv_count = GNRC_NDP_MAX_INIT_RTR_ADV_NUMOF;
                 gnrc_ndp_router_retrans_rtr_adv(entry);
```

Comment:
Still double router advertisements for border router. These lines must be the else case for the if-statement below.